ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/sql/util/DbDispatcher.scala

```diff
@@ -9,51 +9,36 @@ import java.util.concurrent.Executors
 import akka.stream.scaladsl.Source
 import akka.{Done, NotUsed}
 import com.codahale.metrics.MetricRegistry
+import com.digitalasset.ledger.api.health.{HealthStatus, ReportsHealth}
 import com.digitalasset.platform.common.logging.NamedLoggerFactory
 import com.digitalasset.platform.common.util.DirectExecutionContext
 import com.digitalasset.platform.sandbox.stores.ledger.sql.dao.HikariJdbcConnectionProvider
 import com.google.common.util.concurrent.ThreadFactoryBuilder
 
 import scala.concurrent.{ExecutionContext, Future}
 
-trait DbDispatcher extends AutoCloseable {
-
-  /** Runs an SQL statement in a dedicated Executor. The whole block will be run in a single database transaction.
-    *
-    * The isolation level by default is the one defined in the JDBC driver, it can be however overridden per query on
-    * the Connection. See further details at: https://docs.oracle.com/cd/E19830-01/819-4721/beamv/index.html
-    */
-  def executeSql[T](description: String, extraLog: Option[String] = None)(
-      sql: Connection => T): Future[T]
-
-  /**
-    * Creates a lazy Source, which takes care of:
-    * - getting a connection for the stream
-    * - run the SQL query using the connection
-    * - close the connection when the stream ends
-    *
-    * @param sql a streaming SQL query
-    * @tparam T the type of streamed elements
-    * @return a lazy source which will only access the database after it's materialized and run
-    */
-  def runStreamingSql[T](sql: Connection => Source[T, Future[Done]]): Source[T, NotUsed]
-
-  /** The number of pre-allocated connections for short lived queries */
-  def noOfShortLivedConnections: Int
-}
-
-private class DbDispatcherImpl(
+/**
+  * A helper class to dispatch blocking SQL queries onto a dedicated thread pool. The number of threads are being kept
+  * * in sync with the number of JDBC connections in the pool.
+  *
+  * @param jdbcUrl                   the jdbc url containing the database name, user name and password
+  * @param noOfShortLivedConnections the number of connections to be pre-allocated for regular SQL queries
+  * @param noOfStreamingConnections  the max number of connections to be used for long, streaming queries
+  */
+class DbDispatcher(
```

Comment:
```suggestion
final class DbDispatcher(
```
If it's safe to close the hierarchy here, I'd suggest we do so.